openhtf/plugs/__init__.py

```diff
@@ -376,15 +379,21 @@ def _initialize_rpc_server(self):
       server_thread.start()
       self._xmlrpc_server = server
 
-  def initialize_plugs(self, plug_types=None):
+  def initialize_plugs(self, plug_types=None, partial=False):
     """Instantiate required plugs.
 
-    Instantiates known plug types and saves the instances in self._plugs_by_type
-    for use in ProvidePlugs().  Additional plug types may be specified here
-    rather than passed into the constructor (this is used primarily for unit
-    testing phases).
+    Instantiates plug types and saves the instances in self._plugs_by_type for
+    use in provide_plugs().
+
+    Args:
+      plug_types: Additional plug types may be specified here rather than passed
+                  into the constructor (this is used primarily for unit testing
+                  phases).
+      partial: If true, _only_ instantiate the plugs named in plug_types. Skip
```

Comment:
blech, passing this as an arg is pretty awkward.  how about making this method always only instantiate the provided ones, and if not provided, default to instantiating `self._plug_types`?  I don't think anywhere relies on the union behavior, but if it does, it should explicitly call `initialize_plugs()` first then initialize the extra ones.

edit: see below, maybe refactor this a little instead.